### ISSUE TITLE
[netcore] Make System.Globalization.Tests.RegionInfoPropertyTests.EnglishName Pass

### ISSUE THREAD

**Issue by steveisok** [mono/mono#15354]
The test currently fails with:

```
System.Globalization.Tests.RegionInfoPropertyTests.EnglishName(name: "zh-CN", expected: ["China", "People's Republic of China"]) [FAIL]
      Assert.True() Failure
      Expected: True
      Actual:   False
      Stack Trace:
           at System.Globalization.Tests.RegionInfoPropertyTests.EnglishName(String name, String[] expected)
           at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    System.Globalization.Tests.RegionInfoPropertyTests.EnglishName(name: "CN", expected: ["China", "People's Republic of China"]) [FAIL]
      Assert.True() Failure
      Expected: True
      Actual:   False
      Stack Trace:
           at System.Globalization.Tests.RegionInfoPropertyTests.EnglishName(String name, String[] expected)
           at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

Determine why and make the test pass.

**Comment by fanyang-mono** [comment]
With net core build, RegionInfo("zh-CN").EnglishName returns "China mainland". However, the test expects "China" or "People's Republic of China". That's why the test failed.

With default mono build, RegionInfo("zh-CN").EnglishName returns "China".

**Comment by lambdageek** [comment]
Looks like CoreFX fixed this test upstream: https://github.com/dotnet/corefx/pull/38577

**Comment by fanyang-mono** [comment]
For mono with net core, RegionInfo class is defined in netcore/System.Private.CoreLib/shared/System/Globalization/RegionInfo.cs

RegionInfo("zh-CN").EnglishName gets value from its private property _cultureData.EnglishCountryName, which is an instance of class CultureData.

In CultureData.cs, the getter function of EnglishCountryName calls into GetLocaleInfo(LocaleStringData type), which calls into GetLocaleInfo(string localeName, LocaleStringData type), which calls into Interop.Globalization.GetLocaleInfoString.

In netcore/System.Private.CoreLib/shared/Interop/Unix/System.Globalization.Native/Interop.Locale.cs, GetLocaleInfoString is defined as a function from System.Globalization.Native.dll(or .so). By searching the corresponding entry point() in dotnet/coreclr repository, we found the definition for that here:
https://github.com/dotnet/coreclr/blob/a51af0838709ca81b51bee236865a39b31f68f41/src/corefx/System.Globalization.Native/pal_localeStringData.c#L206

And it eventually calls into uloc_getDisplayCountry.

When setting the following two environment variable, we found that the libraries being loaded were correct as well.
MONO_LOG_LEVEL=debug, MONO_LOG_MASK=dll

Based on what Aleksey found, we don't need to do anything any more. We just need to verify that the test does pass when we get the fix from upstream.
